src/sage/groups/braid.py

```diff
@@ -1614,27 +1680,55 @@ def conjugating_braid(self, other):
 
         INPUT:
 
-        - ``other`` -- the other braid to look for conjugating braid
+        - ``other`` -- a braid in the same braid group as ``self``.
+
+        OUTPUT:
+
+        A braid `d` such that if `o` equals ``other`` and `s` equals ``self``
+        then `o = d^{-1 \cdot s \cdot d`.
 
         EXAMPLES::
 
             sage: B = BraidGroup(3)
             sage: a = B([2, 2, -1, -1])
             sage: b = B([2, 1, 2, 1])
             sage: c = b * a / b
-            sage: d = a.conjugating_braid(c)
-            sage: d * c / d == a
-            True
-            sage: d
+            sage: d1 = a.conjugating_braid(c)
+            sage: print(d1)
             s1*s0
-            sage: d * a / d == c
+            sage: d1 * c / d1 == a
+            True
+            sage: d1 * a / d1 == c
+            False
+            sage: l = sage.groups.braid.conjugatingbraid(a,c)
+            sage: d1 == B._element_from_libbraiding(l)
+            True
+            sage: b = B([2, 2, 2, 2, 1])
+            sage: c = b * a / b
+            sage: d1 = a.conjugating_braid(c)
+            sage: print(len(d1.Tietze()))
+            7
+            sage: d1 * c / d1 == a
+            True
+            sage: d1 * a / d1 == c
             False
+            sage: d1
+            s1^2*s0^2*s1^2*s0
+            sage: l = sage.groups.braid.conjugatingbraid(a,c)
+            sage: d2 = B._element_from_libbraiding(l)
+            sage: print(len(d2.Tietze()))
+            13
+            sage: c.conjugating_braid(b) is None
+            True
         """
         l = conjugatingbraid(self, other)
         if not l:
             return None
         else:
-            return self.parent()._element_from_libbraiding(l)
+            B = self.parent()
+            n = B.strands()
```

Comment:
what is the reason for getting the number of strands?

Reply:
To define the symmetric group

Reply:
You no longer define it here.